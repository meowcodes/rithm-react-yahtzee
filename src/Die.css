.Die {
  display: inline-block;
  padding: 0em;
  background-color: red;
  color: white;
  margin-right: 1em;
  border-radius: 20px;
  border: 0px;
  outline:none;
}

.Die-locked {
  background-color: darkred;
}

.Die-image {
  width: 150px;
}