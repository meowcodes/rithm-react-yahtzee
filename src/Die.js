import React, { Component } from "react";
import "./Die.css";
import die1 from "./die-1.svg";
import die2 from "./die-2.svg";
import die3 from "./die-3.svg";
import die4 from "./die-4.svg";
import die5 from "./die-5.svg";
import die6 from "./die-6.svg";

class Die extends Component {
  constructor(props) {
    super(props);
    this.state = {
      images: [die1, die2, die3, die4, die5, die6]
    }
  }

  render() {
    return (
      <button
        className={this.props.locked ? "Die Die-locked" : "Die"}
        onClick={() => this.props.handleClick(this.props.idx)}>
        <img className="Die-image" src={this.state.images[this.props.val - 1]} />
      </button>
    );
  }
}

export default Die;
