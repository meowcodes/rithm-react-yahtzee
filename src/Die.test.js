import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import Die from "./Die";

// smoke test
it("renders without crashing", function () {
    mount(<Die />);
});

// snapshot test - unlocked
it("matches snapshot", function () {
    let wrapper = mount(<Die
        val={3} />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});

// snapshot test - locked
it("matches snapshot", function () {
    let wrapper = mount(<Die
        val={3}
        locked={true} />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});