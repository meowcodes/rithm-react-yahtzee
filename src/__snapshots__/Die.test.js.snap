// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Die
  val={3}
>
  <button
    className="Die"
    onClick={[Function]}
  >
    <img
      className="Die-image"
      src="die-3.svg"
    />
  </button>
</Die>
`;

exports[`matches snapshot 2`] = `
<Die
  locked={true}
  val={3}
>
  <button
    className="Die Die-locked"
    onClick={[Function]}
  >
    <img
      className="Die-image"
      src="die-3.svg"
    />
  </button>
</Die>
`;
